### PR TITLE
FIX Check whether user is eligible for MFA in canSkipMFA method

### LINKED ISSUE #376: (CMS3) Users with no CMS access can't authenticate when MFA is mandatory
#### Affects

~MFA 4.0.x~, 3.0.x

#### Description

Following on from #369, we've received and corroborated a report that Members with no CMS access are barred from completing login when MFA is configured to be mandatory. This is due to a discrepancy in behaviour between the methods used to determine whether a user can bypass MFA at different stages of the login process:

- When determining whether to prompt for MFA, [`EnforcementManager::shouldRedirectToMFA()`](https://github.com/silverstripe/silverstripe-mfa/blob/fb7849fb752a229a4b2968c04b88651c6fccfe89/src/Service/EnforcementManager.php#L95) is called, which determines whether the user has access to the CMS and returns `false`, allowing login to proceed.
- Later, after login has completed, [`MFA\MemberAuthenticator\LoginHandler::redirectAfterSuccessfulLogin()`](https://github.com/silverstripe/silverstripe-mfa/blob/fb7849fb752a229a4b2968c04b88651c6fccfe89/src/Authenticator/LoginHandler.php#L432) calls [`EnforcementManager::canSkipMFA()`](https://github.com/silverstripe/silverstripe-mfa/blob/fb7849fb752a229a4b2968c04b88651c6fccfe89/src/Service/EnforcementManager.php#L63), which _does not_ check whether the user has access to the CMS and therefore returns `false`, causing an immediate logout and redirect to the authentication screen.

#### Replication

- Create test Member with no CMS permissions
- Set MFA to be mandatory in Settings
- Attempt to log in as test Member, observe immediate redirect back to login

#### PRs

- [x] 4.0 branch: #379 
- [x] 3.0 branch: #382

### PR DESCRIPTION
**Pull request by Cheddam**
Users with no CMS access are not eligible to use MFA by default. This was respected in the shouldRedirectToMFA method, but not in canSkipMFA, which resulted in these users being booted back to the login screen whenever they attempted to log in.

This commit also tidies some of the docblocks in EnforcementManager, and adjusts the hasAdminAccess method to always act as the provided Member.

Will resolve #376 for the 4.0 branch; requires backporting to 3.0 alongside #369.

### CODE REVIEW

**Comment by Cheddam** [comment]
Pulled this one out of the oven a little early 😅

Working on expanding the unit test coverage for canSkipMFA to include more states.

**Comment by Cheddam** [comment]
Tests are green, and I've successfully manually tested the fix. The change is not hitting the coverage goal for the diff, despite bumping it for the overall project. I've added additional test coverage to get across as many code paths as possible, but...

Coverage has been 'lost' on [the loop that checks LeftAndMain menu items for access](https://codecov.io/gh/silverstripe/silverstripe-mfa/pull/379/src/src/Service/EnforcementManager.php#L235). I'm actually fairly confident this code _never_ runs - [`LeftAndMain::MainMenu()`](https://github.com/silverstripe/silverstripe-admin/blob/1/code/LeftAndMain.php#L1004) calls [`CMSMenu::get_viewable_menu_items()`](https://github.com/silverstripe/silverstripe-admin/blob/1/code/CMSMenu.php#L252), which has its own internal canView checks and seems to return an empty ArrayList for members with no admin access. So.. we might be able to drop the loop and just check if it's empty? cc/ @robbieaverill 

Basically, the PR is green, I've bumped overall test coverage slightly in the process, and codecov is being unreasonable. Ready for a peer review! 👍

File: tests/php/Service/EnforcementManagerTest.php [hunk 1]
```diff
@@ -84,12 +111,14 @@ public function testShouldNotRedirectToMFAWhenUserDoesNotHaveCMSAccess()
...
     public function testShouldRedirectToMFAWhenUserDoesNotHaveCMSAccessButTheCheckIsDisabledWithConfig()
     {
+        Config::nest();
         EnforcementManager::config()->set('requires_admin_access', false);
 
         /** @var Member $member */
         $member = $this->objFromFixture(Member::class, 'sammy_smith');
         $this->logInAs($member);
         $this->assertTrue(EnforcementManager::create()->shouldRedirectToMFA($member));
+        Config::unnest();
```
Comment (robbieaverill) [inline tests/php/Service/EnforcementManagerTest.php]: You shouldn’t need these, SapphireTest does it for you

**Review by robbieaverill — COMMENTED** [review]
Nice one! Looks good, I haven’t tested it though. I’d suggest leaving the loop there but being okay with it being untested.

**Review by dhensby — APPROVED** [review]
